They rejected upstream: https://github.com/triton-lang/triton/pull/9515
We should remove this patch as soon as we upgraded all our targets to cuda 12.9+.

--- a/third_party/nvidia/backend/compiler.py
+++ b/third_party/nvidia/backend/compiler.py
@@ -502,7 +502,11 @@
 
             # Use -Ofc mid to compile ConSan code, if nothing else is specified.
             if any(mode in knobs.compilation.instrumentation_mode for mode in ["consan", "fpsan"]):
-                ptx_extra_options += ["-Ofc", "mid"]
+                ptxas_version_str = get_ptxas(self.target.arch).version
+                ptxas_version = tuple(map(int, ptxas_version_str.split('.')[:2]))
+                if ptxas_version > (12, 8):
+                    # 12.9+ / 13.x fully supports the 'mid' optimization profile
+                    ptx_extra_options += ["-Ofc", "mid"]
 
             # Add --regAllocOptLevel=2 to work around ptxas 13.x bug
             reg_alloc = ['--regAllocOptLevel=2']
